spec/std/float_printer/ryu_printf_spec.cr

```diff
@@ -0,0 +1,3187 @@
+  {1e-134, 498, 364},
+  {1e-133, 493, 360},
+  {1e-132, 487, 354},
+  {1e-131, 486, 354},
+  {1e-130, 482, 352},
+  {1e-129, 477, 347},
+  {1e-128, 478, 350},
+  {1e-127, 473, 346},
+  {1e-126, 470, 343},
+  {1e-125, 468, 343},
+  {1e-124, 464, 339},
+  {1e-123, 459, 336},
+  {1e-122, 458, 336},
+  {1e-121, 452, 330},
+  {1e-120, 451, 330},
+  {1e-119, 448, 329},
+  {1e-118, 444, 325},
+  {1e-117, 441, 324},
+  {1e-116, 437, 320},
+  {1e-115, 431, 316},
+  {1e-114, 430, 316},
+  {1e-113, 428, 314},
+  {1e-112, 424, 311},
+  {1e-111, 421, 310},
+  {1e-110, 417, 307},
+  {1e-109, 414, 304},
+  {1e-108, 411, 303},
+  {1e-107, 407, 300},
+  {1e-106, 405, 298},
+  {1e-105, 401, 295},
+  {1e-104, 398, 293},
+  {1e-103, 395, 291},
+  {1e-102, 391, 288},
+  {1e-101, 388, 287},
+  {1e-100, 381, 281},
+  {1e-99, 380, 281},
+  {1e-98, 378, 279},
+  {1e-97, 375, 278},
+  {1e-96, 370, 273},
+  {1e-95, 368, 272},
+  {1e-94, 364, 269},
+  {1e-93, 360, 266},
+  {1e-92, 358, 265},
+  {1e-91, 355, 264},
+  {1e-90, 351, 260},
+  {1e-89, 348, 259},
+  {1e-88, 345, 256},
+  {1e-87, 342, 255},
+  {1e-86, 338, 252},
+  {1e-85, 334, 248},
+  {1e-84, 328, 244},
+  {1e-83, 327, 244},
+  {1e-82, 320, 237},
+  {1e-81, 319, 237},
+  {1e-80, 318, 237},
+  {1e-79, 313, 233},
+  {1e-78, 312, 233},
+  {1e-77, 306, 228},
+  {1e-76, 305, 228},
+  {1e-75, 299, 223},
+  {1e-74, 298, 223},
+  {1e-73, 295, 221},
+  {1e-72, 291, 218},
+  {1e-71, 287, 215},
+  {1e-70, 285, 214},
+  {1e-69, 280, 210},
+  {1e-68, 278, 210},
+  {1e-67, 275, 207},
+  {1e-66, 271, 204},
+  {1e-65, 268, 202},
+  {1e-64, 265, 200},
+  {1e-63, 262, 199},
+  {1e-62, 255, 193},
+  {1e-61, 254, 193},
+  {1e-60, 251, 190},
+  {1e-59, 245, 186},
+  {1e-58, 244, 186},
+  {1e-57, 242, 184},
+  {1e-56, 238, 182},
+  {1e-55, 235, 179},
+  {1e-54, 230, 176},
+  {1e-53, 229, 176},
+  {1e-52, 221, 169},
+  {1e-51, 220, 169},
+  {1e-50, 219, 169},
+  {1e-49, 215, 165},
+  {1e-48, 209, 160},
+  {1e-47, 208, 160},
+  {1e-46, 205, 159},
+  {1e-45, 202, 156},
+  {1e-44, 199, 154},
+  {1e-43, 195, 152},
+  {1e-42, 192, 150},
+  {1e-41, 189, 148},
+  {1e-40, 183, 142},
+  {1e-39, 182, 142},
+  {1e-38, 177, 138},
+  {1e-37, 175, 138},
+  {1e-36, 171, 134},
+  {1e-35, 169, 134},
+  {1e-34, 165, 130},
+  {1e-33, 160, 127},
+  {1e-32, 159, 127},
+  {1e-31, 148, 117},
+  {1e-30, 147, 117},
+  {1e-29, 149, 119},
+  {1e-28, 146, 117},
+  {1e-27, 138, 111},
+  {1e-26, 137, 111},
+  {1e-25, 136, 111},
+  {1e-24, 132, 107},
+  {1e-23, 129, 105},
+  {1e-22, 125, 103},
+  {1e-21, 122, 100},
+  {1e-20, 119, 98},
+  {1e-19, 114, 94},
+  {1e-18, 110, 92},
+  {1e-17, 109, 92},
+  {1e-16, 104, 87},
+  {1e-15, 101, 86},
+  {1e-14, 99, 84},
+  {1e-13, 95, 82},
+  {1e-12, 92, 79},
+  {1e-11, 89, 77},
+  {1e-10, 86, 76},
+  {1e-9, 82, 73},
+  {1e-8, 78, 70},
+  {1e-7, 73, 65},
+  {1e-6, 72, 65},
+  {1e-5, 69, 64},
+  {1e-4, 66, 62},
+  {1e-3, 60, 57},
+  {1e-2, 59, 57},
+  {1e-1, 55, 54},
+  {1e0, 0, 1}, # TODO: fix exp_reference
```

Comment:
**question:** I don't quite understand what this TODO comment means. Could you enhance it a bit?